lib/dalli/client.rb

```diff
@@ -292,7 +295,9 @@ def groups_for_keys(*keys)
     end
 
     def mapped_keys(keys)
-      keys.flatten.map {|a| validate_key(a.to_s)}
+      keys_array = keys.flatten
+      keys_array.map! { |a| validate_key(a.to_s) }
+      keys_array
```

Comment:
why need to use `keys_array `?
`#map!` returns the same:

```
> x = [1, 2, 3]
 => [1, 2, 3] 
> x.map! { |i| i * 2  }
 => [2, 4, 6] 
```

Reply:
[].map! returns nil

Reply:
![screen shot 2018-10-24 at 10 41 11](https://user-images.githubusercontent.com/422109/47414393-865c2300-d779-11e8-98b5-cc03084b839a.PNG)

hm, strange. In which version of ruby?

Reply:
I was on mobile so I ran it in the browser. Guess it has a different behavior. Ignore my previous comment.